test/eventHandlers/WorkerInitHandler.test.ts

```diff
@@ -117,25 +117,125 @@ describe('WorkerInitHandler', () => {
-    });
+        async function verifyInitFails(): Promise<void> {
+            const fileSubpath = await setTestAppMainField('throwError.js');
+            stream.addTestMessage(msg.init.request(testAppPath));
+            const errorMessage = `Worker was unable to load entry point "${fileSubpath}": test`;
+            await stream.assertCalledWith(
+                msg.init.receivedRequestLog,
+                msg.loadingEntryPoint(fileSubpath),
+                msg.init.failedResponse(errorMessage)
+            );
+        }
 
-    it('Fails for invalid entry point', async function (this: ITestCallbackContext) {
-        const fileSubpath = await setTestAppMainField('throwError.js');
+        describe('Node >=v20', () => {
+            before(function (this: IHookCallbackContext) {
+                if (!isNode20Plus()) {
+                    this.skip();
+                }
+            });
+
+            it('Fails for missing entry point file', async () => {
+                const fileSubpath = await setTestAppMainField('missing.js');
+
+                stream.addTestMessage(msg.init.request(testAppPath));
+                const message = `Worker was unable to load entry point "${fileSubpath}": File does not exist`;
+                await stream.assertCalledWith(
+                    msg.init.receivedRequestLog,
+                    msg.errorLog(message),
+                    msg.init.failedResponse(message)
+                );
+            });
+
+            it('Fails for missing entry point glob pattern', async () => {
+                const fileSubpath = await setTestAppMainField('missing/*.js');
+
+                stream.addTestMessage(msg.init.request(testAppPath));
+                const message = `Worker was unable to load entry point "${fileSubpath}": Found zero files matching the supplied pattern`;
+                await stream.assertCalledWith(
+                    msg.init.receivedRequestLog,
+                    msg.errorLog(message),
+                    msg.init.failedResponse(message)
+                );
+            });
+
+            it('init fails (v3)', async function (this: ITestCallbackContext) {
+                await verifyInitFails();
+            });
+
+            it('init fails (v4) (app setting=0)', async () => {
+                worker.app.programmingModel = <any>{ name: '@azure/functions', version: '4.0.0' };
+                process.env.FUNCTIONS_NODE_BLOCK_ON_ENTRY_POINT_ERROR = '0';
+                await verifyInitFails();
+            });
+
+            it('init fails (v4)', async () => {
+                worker.app.programmingModel = <any>{ name: '@azure/functions', version: '4.0.0' };
+                await verifyInitFails();
+            });
+
+            it('init fails (v3) (app setting=1)', async () => {
+                process.env.FUNCTIONS_NODE_BLOCK_ON_ENTRY_POINT_ERROR = '1';
+                await verifyInitFails();
+            });

```

Comment:
Is there a reason we're not testing the cases with v3 app setting = 0, and v4 app setting =1?

Reply:
Because the app setting wouldn't change the behavior in that case

Reply:
I don't get what you're saying. It _shouldn't_ change the behavior, but don't we need to check it actually _doesn't_? You could also say v3/v4/app setting=0/1 shouldn't change the behavior if it's node >=20 anyway

Reply:
I try not to overthink unit tests in general - a bunch of this is super arbitrary. That's why my response was short, but I can explain more

First of all, we don't _need_ to test any of this. A lot of times the tests I choose to add are because I use [test-driven-development](https://en.wikipedia.org/wiki/Test-driven_development). If you're not familiar, that means I'll add tests before I write the source code, and ideally they fail until I actually implement the feature so that I know my code made a difference. My previous comment was alluding to the fact that your suggested test would've succeeded before I implemented any of the app setting logic, not really helping with my test-driven-development.

Second, I guess I figure your suggestion would be testing `isEnvironmentVariableSet` itself, which should theoretically be covered by its own unit tests instead of these unit tests.

Lastly, I copied all the node <= 20 tests to use for node >= 20, so I didn't really analyze each one individually. I somewhat consider unit tests as a spec for a feature, and when it comes to breaking changes, I like the idea of copying because you're essentially taking the previous spec and saying "okay here's what it does now"